Add parser tests [WIK-724]

DanielButAtWork opened the pull request:
Adds 2 tests against mediawiki.org
- check that the version starts with `1.`
- check that the version is exactly `1.41`

The first one passes but isn't very useful, the second one doesn't pass because the version is actually (currently) 1.41.0.13 and will change each week - we need to figure out how to make the test accept the version as a general pattern instead of needing to hard-code the exact version.

DanielButAtWork committed:
Start adding parser tests

Adds 2 tests against mediawiki.org
- check that the version starts with `1.`
- check that the version is exactly `1.41`

The first one passes but isn't very useful, the second one doesn't pass because
the version is actually (currently) 1.41.0.13 and will change each week - we
need to figure out how to make the test accept the version as a general pattern
instead of needing to hard-code the exact version.

DanielButAtWork commented:
Poke (GitHub equivalent of) Jenkins - rerun tests following WikiTeq/mediawiki-phpunit-action#1

DanielButAtWork committed:
parser tests: only test that function exists

Parser tests cannot easily test that the output of remote_version matches a
regex, so just test that the parser function is usable and returns a version
starting with `1.`.
WIK-724

vedmaka commented on tests/parser/parserTests.txt:
```diff
@@ -0,0 +1,17 @@
+mediawiki.org version - starts with 1.
+Used to ensure that the parser hook is registered
+!! wikitext
+{{padleft:|2|{{#remote_version:https://www.mediawiki.org/w/api.php}}}}
```
I feel that this is more of an integration test than a parser test and it doesn't seem right. In my opinion, it doesn't make much sense to use parser tests to test the extension parser functions. Therefore, I think it would be better if we didn't have parser tests for these functions at all

DanielButAtWork committed:
Parser tests: remove `functionhooks` usage

If `functionhooks` are used to verify that the extension is installed, then
when the extension is not installed the tests are skipped and pass rather than
fail as desired. Thus, just do not check that the extension is loaded, and if
it is not the test will fail.

WIK-724
